cmake/scripts/parse_cargo_packages.py

```diff
@@ -0,0 +1,56 @@
+
+import os
+import re
+import sys
+
+
+def parse_members_for_workspace(toml_path):
+    with open(toml_path, mode='rb') as f:
+        data = f.read()
+
+    manifest = data.decode('utf8')
+
+    regex = re.compile(r"^members\s*=\s*\[(.*?)\]", re.S | re.M)
+    members_block = regex.findall(manifest)[0]
+
+    return re.sub(r'[",]', '', members_block).split()
+
+
+def parse_package_name(package_toml_path):
+    with open(package_toml_path, mode='rb') as f:
+        data = f.read()
+
+    manifest = data.decode('utf8')
+
+    regex = re.compile(r'^name\s*=\s*"(.*?)"', re.M)
+
+    return regex.findall(manifest)[0]
+
+
+if len(sys.argv) < 3:
+    err = f"[usage] python3 {sys.argv[0]} cargo_toml_path workspace_path"
+    raise ValueError(err)
+
+toml_path = sys.argv[1]
+workspace_path = sys.argv[2]
+
+backlist = [
+    "mesatee_sdk",
+    "mesatee_sdk_c",
+    "protected_fs_rs",
+    "mesatee_cli",
+]
```

Comment:
`backlist` -> `blacklist`

More importantly, you are involving another place to modify when adding a new app/lib, which is against the intention of this pr.

One possible solution is to distinguish apps from libs by adding a comment `# lib` after lib entries in `toolchain_deps/Cargo.unix_app.toml`.
```
members = [
  "mesatee_config/config_gen",
  "tests/integration_test",
  "mesatee_sdk", # lib
```
If you decide to take this approach, please add some explanation before
```
members = [
```

Reply:
The 2nd approach has been implemented~ @rdzhou 

I think it would be better put all unix libraries into an individual `Cargo.unix_lib.toml` later 😄 .